iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -312,33 +314,43 @@ private static final class IcebergRecordReader<T> extends RecordReader<Void, T>
     private CloseableIterator<T> currentIterator;
     private Table table;
     private boolean fetchVirtualColumns;
+    private boolean isMerge = false;
+    private IcebergMergeSplit mergeSplit;
```

Comment:
Please create a new class like `IcebergMergeRecordReader` and move everything related to merge into that class.
You can extend the existing `IcebergRecordReader` if you can reuse the code and override methods if necessary

Reply:
Created a new record reader impl.